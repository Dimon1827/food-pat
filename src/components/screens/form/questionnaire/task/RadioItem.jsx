import SelectInput from '../../../../ui/selectInput/SelectInput';

function RadioItem({ control, id, name, type }) {
  return (
    <li className="task__item">
      <SelectInput control={control} id={id} type={type} name={name} />
    </li>
  );
}

export default RadioItem;
