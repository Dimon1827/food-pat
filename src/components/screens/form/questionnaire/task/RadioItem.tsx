import { ICheckBoxItem } from 'types';
import SelectInput from '../../../../ui/selectInput/SelectInput';

function RadioItem({ id, name, type }: ICheckBoxItem) {
  return (
    <li className="task__item">
      <SelectInput id={id} type={type} name={name} />
    </li>
  );
}

export default RadioItem;
