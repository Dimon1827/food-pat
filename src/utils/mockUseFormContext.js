import { useFormContext } from "react-hook-form";

export function mockUseFormContext(errors) {
  useFormContext.mockReturnValue({
    register: jest.fn(),
    formState: {
      errors: errors,
    },
  });
}


