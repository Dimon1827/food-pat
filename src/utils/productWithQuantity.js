const productWithQuantity = (food) => {
  return {
    ...food,
    isChecked: false,
    quantity: 1,
  };
};

export default productWithQuantity;
