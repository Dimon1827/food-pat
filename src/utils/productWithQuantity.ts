import { IProducts } from "redux/slice/productsSlice.types";

interface IproductWithQuantity extends Omit<IProducts, 'isChecked' | 'quantity'> {};

const productWithQuantity = (food: IproductWithQuantity) => {
  return {
    ...food,
    isChecked: false,
    quantity: 1,
  };
};

export default productWithQuantity;
