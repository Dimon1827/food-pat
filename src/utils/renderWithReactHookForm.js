import { render } from '@testing-library/react';
import { FormProvider, useForm } from 'react-hook-form';

export function renderWithReactHookForm(ui, { defaultValues = {} } = {}) {
  let reactHookFormMethods = {};

  const Wrapper = ({ children }) => {
    const methods = useForm({ defaultValues });
    return <FormProvider {...methods}>{children}</FormProvider>;
  };

  return {
    ...render(ui, { wrapper: Wrapper }),
  };
}