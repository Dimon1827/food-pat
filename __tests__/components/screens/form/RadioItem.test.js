import { screen } from '@testing-library/react';
import { renderWithReactHookForm } from '@/utils/renderWithReactHookForm';
import RadioItem from '@/components/screens/form/questionnaire/task/RadioItem';

const inputElement = {
    id: 'slimming',
    type: 'radio',
    name: 'Похудение',
}

describe('RadioItem', () => {
  it('RadioItem renders correctly as radio', () => {
    renderWithReactHookForm(<RadioItem {...inputElement} />);
    const input = screen.getByRole('radio');
    const listItem = screen.getByRole('listitem');
    const label = screen.getByText('Похудение');
    expect(listItem).toMatchSnapshot();
    expect(input).toBeInTheDocument();
    expect(label).toBeInTheDocument();
    expect(input).toHaveClass('task__input');
    expect(label).toHaveClass('task__label');
  });
});
