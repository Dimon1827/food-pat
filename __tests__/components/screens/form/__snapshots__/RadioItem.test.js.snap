// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioItem RadioItem renders correctly as radio 1`] = `
<li
  class="task__item"
>
  <input
    class="visually-hidden control-input task__input"
    id="slimming"
    name="task"
    type="radio"
    value="slimming"
  />
  <label
    class="label task__label"
    for="slimming"
  >
    Похудение
  </label>
</li>
`;
